core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -420,7 +427,8 @@ private void wrapEntryForReplaceIfNeeded(InvocationContext ctx, ReplaceCommand c
       // locally, but if there's no need to get remote, the read-only
       // function needs to be executed, so force a non-null entry in
       // context with null content
-      if (entry == null && cdl.localNodeIsOwner(command.getKey())) {
+      // TODO: move this logic to entryFactory
+      if (entry == null && shouldRead(command.getKey())) {
          entryFactory.wrapEntryForReading(ctx, command.getKey(), NullCacheEntry.getInstance());
       }
 
```

Comment:
Err, I need to remove the `ctx.onReturn(commitEntriesReturnHandler);` - as entries are not fetched from remote nodes, there's nothing to commit.